zerver/lib/actions.py

```diff
@@ -4562,8 +4595,27 @@ def user_info(um: UserMessage) -> Dict[str, Any]:
             # Remove duplicates by excluding the id of users already in users_to_be_notified list.
             # This is the case where a user both has a UserMessage row and is a current Subscriber
             subscribers = subscribers.exclude(user_profile_id__in=[um.user_profile_id for um in ums])
+
+            if new_stream is not None:
+                assert guest_subs is not None
+                # Exclude guest users who are not subscribed to the new stream from receing this event.
+                subscribers = subscribers.exclude(user_profile_id__in=[sub.user_profile_id for sub in guest_subs])
+
             # All users that are subscribed to the stream must be notified when a message is edited
             subscribers_ids = [user.user_profile_id for user in subscribers]
+
+            if new_stream is not None:
+                # Get guest users who aren't subscribed to the old stream.
+                # TODO: Guest users don't see the new moved topic unless breadcrumb message for
+                # new stream is enabled. Excluding these users from receiving this event helps
+                # us avoid a error trackeback for our clients. We should figure out a way to
+                # inform the guest users of this new topic if `do_send_message` for these messages
```

Comment:
You don't mean `do_send_messages`, you mean a (new) `message` event, right?

I think it's fine for guests to not see the new messages as a bug we live with for now.

Can guest users see history of public streams they don't have a UserMessage row for?  I think the answer is yes, but we should confirm the history_public_to_subscribers behavior works for them here; otherwise all they will see is the breadcrumb message in any case unless we generate UserMessage rows.

Reply:
Yes, guest users can see all the messages in the new topic, even thought they don't have UserMessage row for it.